lib/filter/tests/test_filters_common.c

```diff
@@ -0,0 +1,259 @@
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filter/filter-expr.h"
+#include "filter/filter-expr-grammar.h"
+#include "filter/filter-netmask.h"
+#include "filter/filter-netmask6.h"
+#include "filter/filter-op.h"
+#include "filter/filter-cmp.h"
+#include "filter/filter-tags.h"
+#include "filter/filter-re.h"
+#include "filter/filter-pri.h"
+#include "cfg.h"
+#include "messages.h"
+#include "syslog-names.h"
+#include "logmsg/logmsg.h"
+#include "apphook.h"
+#include "plugin.h"
+
+
+#include <time.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+MsgFormatOptions parse_options;
+
+gint
+facility_bits(const gchar *fac)
+{
+  return 1 << (syslog_name_lookup_facility_by_name(fac) >> 3);
+}
+
+gint
+level_bits(const gchar *lev)
+{
+  return 1 << syslog_name_lookup_level_by_name(lev);
+}
+
+gint
+level_range(const gchar *from, const gchar *to)
+{
+  int r1, r2;
+
+  r1 = syslog_name_lookup_level_by_name(from);
+  r2 = syslog_name_lookup_level_by_name(to);
+  return syslog_make_range(r1, r2);
+}
+
+FilterExprNode *
+compile_pattern(FilterRE *f, const gchar *regexp, const gchar *type, gint flags)
+{
+  gboolean result;
+
+  log_matcher_options_defaults(&f->matcher_options);
+  f->matcher_options.flags = flags;
+  log_matcher_options_set_type(&f->matcher_options, type);
+
+  result = filter_re_compile_pattern(f, configuration, regexp, NULL);
+
+  if (result)
+    return &f->super;
+
+  filter_expr_unref(&f->super);
+  return NULL;
+}
+
+FilterExprNode *
+create_pcre_regexp_filter(gint field, const gchar *regexp, gint flags)
+{
+  return compile_pattern(filter_re_new(field), regexp, "pcre", flags);
+}
+
+FilterExprNode *
+create_pcre_regexp_match(const gchar *regexp, gint flags)
+{
+  return compile_pattern(filter_match_new(), regexp, "pcre", flags);
+}
+
+LogTemplate *
+create_template(const gchar *template)
+{
+  LogTemplate *t;
+
+  t = log_template_new(configuration, NULL);
+  log_template_compile(t, template, NULL);
+  return t;
+}
+
+#if SYSLOG_NG_ENABLE_IPV6
+static gboolean
+_is_ipv6(const gchar *sockaddr)
+{
+  return (NULL != strchr(sockaddr, ':'));
+}
+#endif
+
+static GSockAddr *
+_get_sockaddr(const gchar *sockaddr)
+{
+  if (!sockaddr)
+    return NULL;
+
+#if SYSLOG_NG_ENABLE_IPV6
+  if (_is_ipv6(sockaddr))
+    {
+      return g_sockaddr_inet6_new(sockaddr, 5000);
+    }
+#endif
+  return g_sockaddr_inet_new(sockaddr, 5000);
+}
+
+void
+testcase_with_socket(const gchar *msg, const gchar *sockaddr,
+                     FilterExprNode *f,
+                     gboolean expected_result)
+{
+  LogMessage *logmsg;
+  gboolean res;
+
+  filter_expr_init(f, configuration);
```

Comment:
The original version asserted for the return value. Please add cr_assert here.

Reply:
cr_assert has been added.